task_sdk/src/airflow/sdk/execution_time/supervisor.py

```diff
@@ -500,6 +508,20 @@ def _monitor_subprocess(self):
 
             self._send_heartbeat_if_needed()
 
+            self._handle_task_overtime_if_needed()
+
+    def _handle_task_overtime_if_needed(self):
+        """Handle termination of auxiliary processes if the task exceeds the configured success overtime."""
+        if self._terminal_state != TerminalTIState.SUCCESS:
+            return
```

Comment:
Why just success? Can't a ti in failed or deferred state also be in overtime/running listeners?

Reply:
(that interface needs changing too so maybe we allow for all terminal conditions)